src/Stratis.Features.FederatedPeg/TargetChain/WithdrawalTransactionBuilder.cs

```diff
@@ -25,28 +38,42 @@ public class WithdrawalTransactionBuilder : IWithdrawalTransactionBuilder
         private readonly IFederationWalletManager federationWalletManager;
         private readonly IFederationWalletTransactionHandler federationWalletTransactionHandler;
         private readonly IFederatedPegSettings federatedPegSettings;
+        private readonly IStateRepositoryRoot stateRepositoryRoot;
 
         public WithdrawalTransactionBuilder(
             ILoggerFactory loggerFactory,
             Network network,
             IFederationWalletManager federationWalletManager,
             IFederationWalletTransactionHandler federationWalletTransactionHandler,
-            IFederatedPegSettings federatedPegSettings)
+            IFederatedPegSettings federatedPegSettings,
+            IStateRepositoryRoot stateRepositoryRoot = null)
```

Comment:
This combined with the check below feels like a combination for weird bugs in the future. We should avoid optional dependencies.

Reply:
Adding this:

```
        ...
        StateRepositoryRoot stateRepositoryRoot = null)
        {
            if (!federatedPegSettings.IsMainChain)
            {
                Guard.NotNull(stateRepositoryRoot, nameof(stateRepositoryRoot));
            }
```